clang/test/SemaSYCL/intel-fpga-loops.cpp

```diff
@@ -216,6 +222,19 @@ void goo() {
   // expected-error@+1 {{'loop_count' attribute requires a non-negative integral compile time constant expression}}
   [[intel::loop_count(-1)]] for (int i = 0; i != 10; ++i)
       a[i] = 0;
+
+  // no diagnostics are expected
+  [[intel::fpga_pipeline]] for (int i = 0; i != 10; ++i)
+    a[i] = 0;
+  // no diagnostics are expected
+  [[intel::fpga_pipeline(0)]] for (int i = 0; i != 10; ++i)
+    a[i] = 0;
+  // no diagnostics are expected
+  [[intel::fpga_pipeline(-1)]] for (int i = 0; i != 10; ++i)
+    a[i] = 0;
```

Comment:
According to previous discussion with here: https://github.com/intel/llvm/pull/4017#discussion_r660808718
The attribute takes Boolean values. 
Any nonzero number should be treated as true, and zero should be treated as false.
So, the attribute accepts non-negative value.

Reply:
In this case it seems to be accepting negative number though.  Should -1 be accepted?

Reply:
According to previous discussion with @AaronBallman here: https://github.com/intel/llvm/pull/4017#discussion_r660808718,  the attribute accepts negative value.

>>Should -1 be accepted?

@AaronBallman, what is your thought?

Reply:
Done. Added IR test